v4/newrelic/segments.go

```diff
@@ -295,7 +319,15 @@ func (s *ExternalSegment) method() string {
 //
 // The key must contain fewer than than 255 bytes.  The value must be a
 // number, string, or boolean.
-func (s *MessageProducerSegment) AddAttribute(key string, val interface{}) {}
+func (s *MessageProducerSegment) AddAttribute(key string, val interface{}) {
```

Comment:
This does feel like a lot of repeated code.  Is there any way we can make the different segment types inherit from a single `Segement` type, maybe via embedding it, or something? I don't know if that would break anything, but it might allow us to write this `AddAttribute` code only once.

Reply:
The imbedded type would need to have access to the `StartTime`, which means that field I think would need to be on the imbedded type, and that's not possible because its changing the API. instead, we could create a single method that all of these `AddAttribute` methods call into, but even that seemed heavy handed to me. I decided to go with this because it was very explicit. If you wanted to know how `AddAttribute` works, you don't need to jump around to a million places just to figure it out. It ends up being a line or two more code this way in total, but ultimately, I think it's the better option for readability.